### PR TITLE
Check for sync pod readiness correctly

### PR DESCRIPTION
**Pull request by jim-minter**
```release-note
NONE
```

We hit a race[1] where we checked the pod's view of readiness before the controller had acted on a change in goal state (should have been checking ObservedGeneration)

[1] https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_openshift-azure/1531/pull-ci-openshift-openshift-azure-master-upgrade-v4.2/189/build-log.txt

@asalkeld @mjudeikis @ehashman fyi

### CODE REVIEW

File: pkg/cluster/kubeclient/ready.go [hunk 1]
```diff
@@ -40,14 +41,27 @@ func (u *kubeclient) WaitForReadyWorker(ctx context.Context, hostname string) er
...
+
+			_, err = u.client.CoreV1().
 				Services("kube-system").
 				ProxyGet("", "sync", "", "/healthz/ready", nil).
 				DoRaw()
 
 			switch {
 			case err == nil:
-				return true, nil
+				return false, nil
```
Comment (ehashman) [inline pkg/cluster/kubeclient/ready.go]: This switch will now never return `true` and is basically just differentiating between different failure modes... can we simplify it?

**Comment by jim-minter** [comment]
nothing compiles for me today.

File: pkg/cluster/kubeclient/ready.go (hunk 1 above)
Comment (jim-minter) [reply pkg/cluster/kubeclient/ready.go]: yes, thanks, pushed

**Comment by jim-minter** [comment]
/retest
quota

**Comment by ehashman** [comment]
/retest
both quota

**Pull request merged**